hub/core/dataset/dataset.py

```diff
@@ -820,3 +820,16 @@ def __result__(self):
 
     def __args__(self):
         return None
+
+    def append(self, sample: Dict[str, Any], skip_ok: bool = False):
+        if not skip_ok:
+            for k in self.tensors:
+                if k not in sample:
+                    raise TensorDoesNotExistError(k)
```

Comment:
we should have different exceptions for both cases, whether a tensor is missing from the appended sample or whether a tensor was provided that doesn't exist in the dataset.